src/geometry/manifold/algebra/left_invariant_derivation.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import geometry.manifold.derivation_bundle
+import ring_theory.derivation
+
+/-!
+
+# Left invariant derivations
+
+In this file we define the concept of left invariant derivation for a Lie group. The concept is the
+analogous to the more classical concept of left invariant vector fields, and this analogy "commutes"
+with the isomorphism between global derivations and left invariant vector fields.
+
+Moreover we prove that `left_invariant_derivation I G` has the structure of a Lie algebra, hence
+implementing one of the possible definitions of Lie algebra.
+
+-/
+
+noncomputable theory
+
+open_locale lie_group manifold
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+(G : Type*) [topological_space G] [charted_space H G] [monoid G] [has_smooth_mul I G] (g h : G)
+
+-- Generate trivial has_sizeof instance.
```

Comment:
Could you leave a comment explaining what problem this prevents? Given it is weird it needs some explanation.